nightfall/detection_rules.py

```diff
@@ -0,0 +1,253 @@
+    regex: Regex = None
+    word_list: WordList = None
+
+    def __post_init__(self):
+        if (not self.regex and not self.word_list) or (self.regex and self.word_list):
+            raise NightfallUserError("need either regex or word_list to build an ExclusionRule", 40001)
+
+    def as_dict(self):
+        result = {"matchType": self.match_type.value}
+        if self.regex:
+            result["regex"] = self.regex.as_dict()
+            result["exclusionType"] = "REGEX"
+        if self.word_list:
+            result["wordList"] = self.word_list.as_dict()
+            result["exclusionType"] = "WORD_LIST"
+        return result
+
+
+@dataclass
+class MaskConfig:
+    """An object that specifies how findings should be masked when returned by the API.
+    Attributes:
+        masking_char (chr): character that will be repeated to replace the finding.
+            This character may be a multi-byte character, but it must be exactly one codepoint.
+        num_chars_to_leave_unmasked (int): the number of characters to leave unmasked at either the left or right of
+            the finding when it is returned.
+        mask_right_to_left (bool): True if num_chars_to_leave_unmasked should be on the right, False otherwise.
+        chars_to_ignore (List[chr]): the set of characters to leave unmasked when the finding is returned. These
+            characters may be multi-byte characters, but each entry in the array must be exactly one codepoint.
+        """
+    masking_char: chr
+    num_chars_to_leave_unmasked: int = 0
+    mask_right_to_left: bool = False
+    chars_to_ignore: List[chr] = field(default_factory=list)
+
+    def as_dict(self):
+        return {
+            "maskingChar": self.masking_char,
+            "numCharsToLeaveUnmasked": self.num_chars_to_leave_unmasked,
+            "maskRightToLeft": self.mask_right_to_left,
+            "charsToIgnore": self.chars_to_ignore
+        }
+
+@dataclass
+class RedactionConfig:
+    """An object that configures how any detected findings should be redacted when returned to the client. When this
+    configuration is provided as part of a request, exactly one of the four types of redaction should be set.
+    One of mask_config, substitution_phrase, infotype_substitution or public_key is required:
+        - Masking: replacing the characters of a finding with another character, such as '*' or '👀'
+        - Info Type Substitution: replacing the finding with the name of the detector it matched, such
+            as CREDIT_CARD_NUMBER
+        - Substitution: replacing the finding with a custom string, such as "oh no!"
+        - Encryption: encrypting the finding with an RSA public key
+    Attributes:
+        remove_finding (bool): Whether the original finding should be omitted in responses from the API.
+        mask_config (MaskConfig): Build a redaction config with masking.
+        substitution_phrase (str or None): Build a redaction config with info type substitution.
```

Comment:
the doc comment for this and `substitution_phrase` are the same. Should `mask_config`, `substitution_phrase`, and `infotype_substitution` be combined into a single config object?